crm_compassion/models/res_partner.py

```diff
@@ -69,7 +69,7 @@ def _compute_opportunity_count(self):
             operator = "child_of" if partner.is_company else "="
             partner.opportunity_count += self.env["crm.lead"].search_count(
                 [
-                    ("partner_id", operator, partner.id),
+                    ("partner_id", operator, partner.ids[0]),
```

Comment:
As discussed, this is a fix that doesn't address the core issue, let's put a comment for other developers reading this to understand that `ids[0]`it is a way to get an id even in the case of generics NewId.